### PR TITLE
Fix/beefy withdraw

### PR DESCRIPTION
**Pull request by RedVeil**

### CODE REVIEW

File: packages/foundry/script/DeployVaultSystem.s.sol
Comment (RedVeil) [inline packages/foundry/script/DeployVaultSystem.s.sol:1]: ignore this file. this was just how i discovered the issue

File: packages/foundry/src/vault/adapter/beefy/BeefyAdapter.sol [hunk 1]
```diff
@@ -62,6 +64,11 @@ contract BeefyAdapter is AdapterBase, WithRewards {
 
     beefyBalanceCheck = IBeefyBalanceCheck(_beefyBooster == address(0) ? _beefyVault : _beefyBooster);
 
+    IBeefyStrat strat = IBeefyStrat(beefyVault.strategy());
```
Comment (RedVeil) [inline packages/foundry/src/vault/adapter/beefy/BeefyAdapter.sol:67]: Technically beefy could change the strategy but i have never seen that happen. So im still contemplating if we should have the try catch in each preview call or just here once.

File: packages/foundry/src/vault/adapter/yearn/YearnAdapter.sol [hunk 2]
```diff
@@ -102,20 +98,20 @@ contract YearnAdapter is AdapterBase {
     return supply == 0 ? shares : shares.mulDiv(yVault.balanceOf(address(this)), supply, Math.Rounding.Up);
   }
 
-  function previewDeposit(uint256 assets) public view virtual override returns (uint256) {
-    return paused() ? 0 : _convertToShares(assets - 10, Math.Rounding.Down);
+  function previewDeposit(uint256 assets) public view override returns (uint256) {
+    return paused() ? 0 : _convertToShares(assets - 0, Math.Rounding.Down);
```
[the remark this answers is not in the document]
Comment (RedVeil) [reply packages/foundry/src/vault/adapter/yearn/YearnAdapter.sol:102]: Will keep an eye on this one and see if it fails again in a few days cause of new deposits/harvests

File: packages/foundry/src/vault/adapter/beefy/BeefyAdapter.sol (hunk 1 above)
Comment (amatureApe) [reply packages/foundry/src/vault/adapter/beefy/BeefyAdapter.sol:67]: This is something that could change if the underlying protocol moves to a different address. I've seen it happen a few times that required strategy migration.

File: packages/foundry/src/vault/adapter/yearn/YearnAdapter.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (amatureApe) [reply packages/foundry/src/vault/adapter/yearn/YearnAdapter.sol:102]: Yea a bit strange. I don't see why it would do that all of a sudden

**Pull request merged**